File: src/cli-sample/PrintSimple.cs
using System;

namespace Xt {

    public class PrintSimple {

        public static void Main(string[] args) {

            using (XtAudio audio = new XtAudio(null, IntPtr.Zero, null, null)) {
                for (int s = 0; s < XtAudio.GetServiceCount(); s++) {
                    XtService service = XtAudio.GetServiceByIndex(s);
                    for (int d = 0; d < service.GetDeviceCount(); d++)
                            Console.WriteLine(service.GetName() + ": " + service.GetDeviceName(d));
                }
            }
        }
    }
}

File: src/java-sample/com/xtaudio/xt/PrintSimple.java
package com.xtaudio.xt;

public class PrintSimple {

    public static void main(String[] args) {

        try (XtAudio audio = new XtAudio(null, null, null, null)) {
            for (int s = 0; s < XtAudio.getServiceCount(); s++) {
                XtService service = XtAudio.getServiceByIndex(s);
                for (int d = 0; d < service.getDeviceCount(); d++)
                        System.out.println(service.getName() + ": " + service.getDeviceName(d));
            }
        }
    }
}
